### PR TITLE
[primTorch] Enable regex error testing for `cat` and related refs

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #87767
* __->__ #87766
* #87765

cc @ezyang @mruberry @ngimel @Lezcano @peterbell10 @fdrocha

### CODE REVIEW

File: torch/_prims/__init__.py
Comment (nkaretnikov) [inline torch/_prims/__init__.py:1815]: Based on a similar check in core PyTorch.

File: torch/testing/_internal/common_methods_invocations.py
Comment (nkaretnikov) [inline torch/testing/_internal/common_methods_invocations.py:2001]: Split these apart because I wanted to have precise error messages, but also didn't want to have a catch-all `else` branch.

File: torch/testing/_internal/common_methods_invocations.py
Comment (nkaretnikov) [inline torch/testing/_internal/common_methods_invocations.py:1824]: What's our plan for this? Everything allocates/copies, but perhaps it might be important in some cases (when?).

File: torch/testing/_internal/common_methods_invocations.py
Comment (nkaretnikov) [inline torch/testing/_internal/common_methods_invocations.py:1910]: This was failing not because of different dtypes, which is allowed. It was failing because `x` is not a sequence, which is not supported by available overloads. So I removed this error input.

**Review by lezcano — APPROVED** [review]
Nice clean-up! I just left a question for @mruberry, but this looks good. Let's wait for him to review though before merging.

File: torch/_prims/__init__.py
Comment (mruberry) [inline torch/_prims/__init__.py:1806]: `cat` should explicitly inform the caller it doesn't accept an empty list of tensors:

```
RuntimeError: torch.cat(): expected a non-empty list of Tensors
```

File: torch/_prims/__init__.py
Comment (mruberry) [inline torch/_prims/__init__.py:1807]: Combine this enumeration with the enumeration of tensors below

File: torch/_prims/__init__.py
Comment (mruberry) [reply torch/_prims/__init__.py:1806]: Aha, this is checked in the ref -- let's add a comment here, then

File: torch/testing/_internal/common_methods_invocations.py
Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:1824]: That's true, as we don't try to model the memory structure of tensors. This is a fine comment for now.

**Pull request closed without merging**
